Plutarch/Api/V1/Value.hs

```diff
@@ -48,3 +57,102 @@ deriving via
   (DerivePConstantViaNewtype Plutus.Value PValue (PMap PCurrencySymbol (PMap PTokenName PInteger)))
   instance
     PConstantDecl Plutus.Value
+
+-- | Works correctly only on 'normalize'd and sorted values!
+instance PEq PValue where
+  a #== b = pto a #== pto b
+
+instance Semigroup (Term s PValue) where
+  a <> b = unionWith # plam (+) # a # b
+
+instance Monoid (Term s PValue) where
+  mempty = pcon (PValue AssocMap.empty)
+
+-- | Construct a singleton 'PValue' containing only the given quantity of the given currency.
+singleton :: Term (s :: S) (PCurrencySymbol :--> PTokenName :--> PInteger :--> PValue)
+singleton = phoistAcyclic $
+  plam $ \symbol token amount ->
+    punsafeFrom (AssocMap.singleton # symbol #$ AssocMap.singleton # token # amount)
+
+-- | Get the quantity of the given currency in the 'PValue'.
+valueOf :: Term (s :: S) (PValue :--> PCurrencySymbol :--> PTokenName :--> PInteger)
+valueOf = phoistAcyclic $
+  plam $ \value symbol token ->
+    AssocMap.foldAt
+      # symbol
+      # 0
+      # plam (\map -> AssocMap.foldAt # token # 0 # plam pfromData # pfromData map)
+      # pto value
+
+-- | Check if the value is zero.
+isZero :: Term (s :: S) (PValue :--> PBool)
```

Comment:
Aren't we assuming this will never happen? (assuming `unionWith` is changed to ensure it never returns values with 0s in them)

Reply:
I've removed the function to simplify the API, it's no better than `(== mempty)` anyway.